incubator/faucet/client/cli/tx.go

```diff
@@ -30,134 +21,61 @@ func GetTxCmd(storeKey string, cdc *codec.Codec) *cobra.Command {
 		RunE:                       client.ValidateCmd,
 	}
 
-	faucetTxCmd.AddCommand(flags.PostCommands(
-		GetCmdMint(cdc),
-		GetCmdMintFor(cdc),
-		GetCmdInitial(cdc),
-		GetPublishKey(cdc),
-	)...)
+	faucetTxCmd.AddCommand(
+		txMint(),
+		txMintFor(),
+	)
 
 	return faucetTxCmd
 }
 
-// GetCmdWithdraw is the CLI command for mining coin
-func GetCmdMint(cdc *codec.Codec) *cobra.Command {
-	return &cobra.Command{
+// GetCmdMint is the CLI command for mining coin
+func txMint() *cobra.Command {
+	cmd := &cobra.Command{
 		Use:   "mint",
 		Short: "mint coin to sender address",
 		Args:  cobra.MaximumNArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			inBuf := bufio.NewReader(cmd.InOrStdin())
-			cliCtx := context.NewCLIContext().WithCodec(cdc)
-
-			txBldr := auth.NewTxBuilderFromCLI(inBuf).WithTxEncoder(utils.GetTxEncoder(cdc))
-			denom := args[0]
-
-			msg := types.NewMsgMint(cliCtx.GetFromAddress(), cliCtx.GetFromAddress(), denom)
-			err := msg.ValidateBasic()
+			ctx, err := client.GetClientTxContext(cmd)
 			if err != nil {
 				return err
 			}
 
-			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+			denom := args[0]
+			minter := ctx.GetFromAddress()
+			msg := types.NewMsgMint(minter, minter, denom)
+			if err = msg.ValidateBasic(); err != nil {
+				return fmt.Errorf("message validation failed: %w", err)
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(ctx, cmd.Flags(), msg)
 		},
 	}
+	flags.AddTxFlagsToCmd(cmd)
+	return cmd
 }
 
-// GetCmdWithdraw is the CLI command for mining coin
-func GetCmdMintFor(cdc *codec.Codec) *cobra.Command {
+// GetCmdMintFor is the CLI command for mining coin
+func txMintFor() *cobra.Command {
```

Comment:
nit: same here re: naming consistency. Also the comment is different than the fn name :)